packages/camera/camera_avfoundation/ios/Classes/FLTCamMediaSettingsAVWrapper.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import AVFoundation;
+@import Foundation;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * @interface FLTCamMediaSettingsAVWrapper
+ * @abstract An interface for performing media settings operations.
+ *
+ * @discussion
+ * xctest-expectation-checking implementation (`TestMediaSettingsAVWrapper`) of this interface can
+ * be injected into `camera-avfoundation` plugin allowing to run media-settings tests without any
+ * additional mocking of AVFoundation classes.
+ */
+@interface FLTCamMediaSettingsAVWrapper : NSObject
+
+/**
+ * @method lockDevice:error:
+ * @abstract Requests exclusive access to configure device hardware properties.
+ * @param captureDevice The capture device.
+ * @param outError The optional error.
+ * @result A BOOL indicating whether the device was successfully locked for configuration.
+ */
+- (BOOL)lockDevice:(nonnull AVCaptureDevice *)captureDevice error:(NSError **)outError;
```

Comment:
Is `outError` intended to be non-nullable here, despite usual Obj-C convention of this being optional?

Reply:
nullable applied